### PR TITLE
Add service checks

### PR DESCRIPTION
**Pull request by wang-arthur**
- Service checks are supported by the agent's dogstatsd server starting in version 5.2.
- Also fixes a couple bugs in event formatting

### CODE REVIEW

File: lib/statsd.rb [hunk 1]
```diff
@@ -198,6 +210,47 @@ def set(stat, value, opts={})
...
+  def format_service_check(name, status, opts={})
+    sc_string = "_sc|#{name}|#{status}"
+
+    SC_OPT_KEYS.each do |name_key|
+      if opts[name_key[0].to_sym]
+        if name_key[0] == 'tags'
+          tags = opts[:tags]
+          tags.each do |tag|
+            rm_pipes tag
+          end
```
Comment (LeoCavaille) [inline lib/statsd.rb]: `.each` doesn't modify the array in-place here.

**Comment by LeoCavaille** [comment]
Otherwise LGTM!

**Comment by remh** [comment]
@wang-arthur looks like it can't be merged, can you have a look please ?

**Comment by wang-arthur** [comment]
Ah! will take a look -- forgot about this one

File: lib/statsd.rb (hunk 1 above)
Comment (wang-arthur) [inline lib/statsd.rb]: I've rebased, so `rm_pipes` is now in-place

**Pull request merged**